src/BlockTypes/SingleProduct.php

```diff
@@ -12,6 +12,17 @@ class SingleProduct extends AbstractBlock {
 	 */
 	protected $block_name = 'single-product';
 
+	/**
+	 * Enqueue frontend assets for this block, just in time for rendering.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 */
+	protected function enqueue_assets( array $attributes ) {
+		parent::enqueue_assets( $attributes );
+		$product_id = $attributes['productId'];
```

Comment:
How do you feel about adding an extra protection here?

```suggestion
		$product_id = intval( $attributes['productId'] );
```

It's not really necessary because you already cast it to an int inside `hydrate_from_api()`, but I think it's a good practice to cast to the correct type the first time you read an external variable just in case in the future somebody refactors the code and uses that variable somewhere else assuming the type had been verified.

Reply:
I agree with you! Thanks for the feedback 👍